### PR TITLE
Fix cable coil dupe and APC icon oddity

### PR DESCRIPTION
**Pull request by p440**
:cl: p440
fix: Fixed duping cable coils with magic APC terminals
fix: Fixed invalid icon state for empty APCs
/:cl:

_the fine details_
first part fixes this:
no terminal + wirecutters => magic terminal
go up cut the terminal, place a new cable on the floor and repeat for infinite cable coil

when cutting the cable it would call `connect_to_network()` on things attached to the node to confirm the connected powernet before disconnecting from it and deleting itself
the problem is (in apc.dm)
`if(!terminal)
		make_terminal()`
would just happily make the terminal when this happened and from what i can see this call is never actually necessary since now the terminals get made when someone constructs them or during Initialize()
(correct me if i'm wrong)

second part fixes this:
if you remove the electronics from an APC you can still close the door with a crowbar and then interact with the interface, including opening up the wires panel and hacking the wires, despite there being no electronics nor wiring to hack. on top of this the icon doesn't update until you open the wires panel, and then it doesn't update until you open the cell door. 
for clarity:
how the icon behaves currently 
 crowbar >  screwdriver >  screwdriver >  crowbar > 

the changes
 crowbar > 

screwdrivering at this point would just bash it with the screwdriver, as it would do with any other invalid tool. and since there's no electronics, using it with an empty hand just does nothing. using the crowbar just opens it up again.

i discovered this while trying to replicate #24240 (i had no success)

### CODE REVIEW

**Comment by p440** [comment]
wait don't merge this, i forgot a test case and just realized it needs to be fixed

**Comment by p440** [comment]
tldr last minute "optimizations" without testing

File: code/modules/power/apc.dm [hunk 1]
```diff
@@ -99,8 +99,8 @@
 /obj/machinery/power/apc/connect_to_network()
 	//Override because the APC does not directly connect to the network; it goes through a terminal.
 	//The terminal is what the power computer looks for anyway.
-	if(!terminal)
-		make_terminal()
+	//if(!terminal)
+	//	make_terminal() //unnecessary
```
Comment (AnturK) [inline code/modules/power/apc.dm]: Just remove instead of commenting out please.

File: code/modules/power/apc.dm [hunk 2]
```diff
@@ -448,6 +450,8 @@
 				update_icon()
 		else if(emagged)
 			to_chat(user, "<span class='warning'>The interface is broken!</span>")
+		else if(stat & MAINT && opened == 0)
```
Comment (Cyberboss) [inline code/modules/power/apc.dm]: `else if ((stat & MAINT) && !opened)`

File: code/modules/power/apc.dm [hunk 3]
```diff
@@ -630,6 +634,8 @@
 			charging = 0
 			src.update_icon()
 		return
+	if(stat & MAINT && opened == 0) //no board; no interface
```
Comment (Cyberboss) [inline code/modules/power/apc.dm]: Same here

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by p440** [comment]
done

**Review by Cyberboss — APPROVED** [review]
